lib/active_job/queue_adapters/amazon_sqs_adapter.rb

```diff
@@ -54,6 +51,25 @@ def message_attributes(job)
           }
         }
       end
+
+      def deduplication_body(job, body)
+        dedup_keys = job.deduplication_keys if job.respond_to?(:deduplication_keys)
+        dedup_keys ||= Aws::Rails::SqsActiveJob.config.deduplication_keys.map(&:to_s)
+
+        ignored_dedup_key = 'job_id'
+
+        if dedup_keys.include?(ignored_dedup_key)
```

Comment:
It might be better to check for an invalid key upfront, when its set rather then every time its used (raise the error where its being made) - I think you could do that in the setter for deduplication_keys.